productcomments.php

```diff
@@ -55,6 +58,9 @@ public function __construct()
         $this->displayName = $this->trans('Product Comments', [], 'Modules.Productcomments.Admin');
         $this->description = $this->trans('Allow users to post reviews on your products and/or rate them based on specific criteria.', [], 'Modules.Productcomments.Admin');
 
+        $this->langId = Context::getContext()->language->id;
+        $this->shopId = Context::getContext()->shop->id ? Context::getContext()->shop->id : Configuration::get('PS_SHOP_DEFAULT');
```

Comment:
```suggestion
        $this->langId = Context::getContext()->language->id;
        $this->shopId = Context::getContext()->shop->id ? Context::getContext()->shop->id : Configuration::get('PS_SHOP_DEFAULT');
```

have you tried to simply use `$this->context`?

Reply:
In fact, they come from [original ObjectModel code](https://github.com/PrestaShop/productcomments/blob/v5.0.3/ProductComment.php#L485).
I did a try and see no difference, and [found why](https://github.com/PrestaShop/PrestaShop/blob/8.0.x/classes/module/Module.php#L321). So for safety, I'd prefer keep them as they were :)